airflow/providers/docker/operators/docker.py

```diff
@@ -282,13 +282,20 @@ def execute(self, context) -> Optional[str]:
             raise Exception("The 'cli' should be initialized before!")
 
         # Pull the docker image if `force_pull` is set or image does not exist locally
+        # pylint: disable=too-many-nested-blocks
         if self.force_pull or not self.cli.images(name=self.image):
             self.log.info('Pulling docker image %s', self.image)
+            latest_status = {}
             for output in self.cli.pull(self.image, stream=True, decode=True):
                 if isinstance(output, str):
                     self.log.info("%s", output)
                 if isinstance(output, dict) and 'status' in output:
-                    self.log.info("%s", output['status'])
+                    if 'id' in output:
+                        if latest_status.get(output['id']) != output['status']:
+                            self.log.info("%s: %s", output['id'], output['status'])
+                            latest_status[output['id']] = output['status']
```

Comment:
```suggestion
                    if 'id' in output and latest_status.get(output['id']) != output['status']:
                        self.log.info("%s: %s", output['id'], output['status'])
                        latest_status[output['id']] = output['status']
```

Reply:
There's a chance that `latest_status.get(output.get("id")) != output["status"]` will also work

Reply:
This change you proposed may not work, because if this `if` is not matched, it goes to `else` and print the `status`. Then it's no difference from the earlier "spamming" status.

This is why I have the nested `if`s here. may you let me know if this clarification makes sense to you?

Reply:
The maye we should revert the ifs:
```py
if 'id' not in output:
   self.log.info("%s", output['status'])
   continue

output_id = output.get("id")
output_status = output.get("status")
if latest_status.get(output_id) != output_status:
   self.log.info("%s: %s", output_id, output_status)
   latest_status[output_id] = output_status
```
I'm not a fan of too many nested ifs as pylint likes to complain about them 😉

Reply:
Actually we already cannot avoid having to apply `disable=too-many-nested-blocks`, but your suggestion above is definitely fair.

Have addressed it in https://github.com/apache/airflow/pull/12763/commits/23e81de0b7170d6132b183319bcd73729e508a52 (with extremely minor change)